lib/matplotlib/colorbar.py

```diff
@@ -900,6 +900,406 @@ def remove(self):
         fig.delaxes(self.ax)
 
 
+class ColorsquareBase(cm.ScalarMappable):
+    n_rasterize = 50  # rasterize solids if number of colors >= n_rasterize
+    def __init__(self, ax, cmap=None,
+                 norm=None,
+                 alpha=None,
+                 xvalues=None,
+                 yvalues=None,
+                 xboundaries=None,
+                 yboundaries=None,
+                 xticks=None,
+                 yticks=None,
+                 xformat=None,
+                 yformat=None,
+                 drawedges=False,
+                 filled=True,
+                 xlabel='',
+                 ylabel='',
+                 ):
+        #: The axes that this colorbar lives in.
```

Comment:
`ax` is probably fine to stay public (it is public on every `Artist`), but the xvalues, xboundaries, etc can probably be made private.